announce.go

```diff
@@ -28,6 +29,9 @@ type Announce struct {
 	doneVar *stm.Var
 	cancel  func()
 
+	inFlightGetPeers int64
```

Comment:
I think you can use Announce.pending instead of this. I think they're the same thing.

Reply:
I think we should keep inFlightGetPeers it's more explicative

Reply:
It's probably weird that I've used STM in this implementation here, but I don't think the way you've handled this count is idiomatic here. I agree the naming isn't great, but doesn't particularly matter when you break up the get_peers and announce_peers steps.